Code/Framework/AzToolsFramework/Tests/PythonLoaderTests.cpp

```diff
@@ -0,0 +1,126 @@
+        AZStd::string engineRoot{ s_testEnginePath };
+        AZ::IO::FixedMaxPath test3rdPartyRoot = m_tempDirectory.GetDirectoryAsFixedMaxPath() / s_test3rdPartySubPath;
+
+        auto result = AzToolsFramework::EmbeddedPython::PythonLoader::GetPythonExecutablePath(test3rdPartyRoot, engineRoot);
+        AZ::IO::FixedMaxPath expectedPath{ testPython3rdPartyPath };
+
+        EXPECT_TRUE(result == expectedPath);
+    }
+
+
+    TEST_F(AzToolsFrameworkPythonLoaderFixture, TestReadPythonEggLinkPaths_Valid)
+    {
+        // Prepare the test folder and create dummy egg-link files
+        AZ::IO::FixedMaxPath testRelativeSiteLIbsPath = AZ::IO::FixedMaxPath(s_test3rdPartySubPath) / "venv" / s_testEnginePathHashId / LY_PYTHON_SITE_PACKAGE_SUBPATH;
+        AZ::IO::FixedMaxPath testFullSiteLIbsPath = m_tempDirectory.GetDirectoryAsFixedMaxPath() / testRelativeSiteLIbsPath;
+        AZ::IO::FileIOBase::GetInstance()->CreatePath(testFullSiteLIbsPath.String().c_str());
+
+        AZStd::vector<AZStd::string> expectedResults;
+        expectedResults.emplace_back(testFullSiteLIbsPath.LexicallyNormal().Native());
+
+        static constexpr const char* testEggLinkPaths[] = { "/lib/path/one", "/lib/path/two", "/lib/path/three" };
+        size_t testEggLinkPathCount = sizeof(testEggLinkPaths) / sizeof(const char*);
+
+
+        for (size_t i = 0; i < testEggLinkPathCount; ++i)
+        {
+            AZStd::string testEggFileName = AZStd::string::format("test-%d.egg-link", static_cast<int>(i));
+            const char* lineBreak = ((i % 2) == 0) ? "\n" : "\r\n";
+            AZStd::string testEggFileContent = AZStd::string::format("%s%s.", testEggLinkPaths[i], lineBreak);
+            expectedResults.emplace_back(AZStd::string(testEggLinkPaths[i]));
+
+            AZ::IO::FixedMaxPath testEggILeNamePath = testRelativeSiteLIbsPath / testEggFileName;
+            AZ::Test::CreateTestFile(m_tempDirectory, testEggILeNamePath, testEggFileContent);
+        }
+
+        // Test the method
+        AZStd::string engineRoot{ s_testEnginePath };
+        AZ::IO::FixedMaxPath test3rdPartyRoot = m_tempDirectory.GetDirectoryAsFixedMaxPath() / s_test3rdPartySubPath;
+        AZStd::vector<AZStd::string> resultEggLinkPaths;
+        AzToolsFramework::EmbeddedPython::PythonLoader::ReadPythonEggLinkPaths(test3rdPartyRoot, engineRoot, resultEggLinkPaths);
+
+        // Sort the expected and actual lists
+        AZStd::sort(expectedResults.begin(), expectedResults.end());
+        AZStd::sort(resultEggLinkPaths.begin(), resultEggLinkPaths.end());
+
+        size_t expectedSize = expectedResults.size();
+        size_t actualSize = resultEggLinkPaths.size();
+        EXPECT_EQ(expectedSize, actualSize);
+
+        for (size_t i = 0; i < expectedSize; i++)
+        {
+            AZStd::string expectedPath = expectedResults[i];
+            AZStd::string actualPath = resultEggLinkPaths[i];
+
+            EXPECT_TRUE(expectedPath == actualPath);
+        }
```

Comment:
AZStd::vector class has a `operator==` that performs a deep compare of the elements.
So instead of comparing the size and then each element of the vectors, the vectors can just be passed to `EXPECT_EQ`.

```suggestion
        EXPECT_EQ(expectedResults, resultEggLinkPaths);
```